### PR TITLE
WebClient.apiCall() will detect some types of bad arguments

### LINKED ISSUE #521: Fail with a meaningful error when WebClient.apiCall()'s first argument is not a key-value object
### Description

This is useful when a user calls `apiCall()` (or any of the named method aliases) with a flat list of arguments accidentally. This is more likely to happen when a user is migrating from v3 or earlier.

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

### PR DESCRIPTION
**Pull request by aoberoi**
###  Summary

fixes #521

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: src/WebClient.spec.js [hunk 1]
```diff
@@ -100,6 +100,37 @@ describe('WebClient', function () {
...
+        const results = [
+          this.client.apiCall('method', 4),
+          this.client.apiCall('method', 'a string'),
+          this.client.apiCall('method', false),
+        ];
+        const caughtErrors = results.map(r => {
+          assert(isPromise(r));
+          return r
+            .then(() => {
+              // if any of these promises resolve, this test failes
```
Comment (Roach) [inline src/WebClient.spec.js]: Fancy spelling

Comment (aoberoi) [reply src/WebClient.spec.js]: hahaha

File: src/WebClient.ts [hunk 2]
```diff
@@ -121,7 +121,11 @@ export class WebClient extends EventEmitter {
 
     // The following thunk is the actual implementation for this method. It is wrapped so that it can be adapted for
     // different executions below.
-    const implementation = () => {
+    const implementation = async () => {
+
+      if (typeof options === 'string' || typeof options === 'number' || typeof options === 'boolean') {
```
Comment (Roach) [inline src/WebClient.ts]: Could you check for `is an object` rather than `is not x, or x, or x`?

Comment (aoberoi) [reply src/WebClient.ts]: `typeof null === 'object'` so that would pass this check. that being said, it passes this check right now too. 🤔

Comment (aoberoi) [reply src/WebClient.ts]: im afraid of weird javascript quirks, and this is more explicit regardless. `null`s would still pass this check, but `null` never has been a valid argument value, so i'm not too afraid of that case. i stand by this condition.

**Pull request merged**